internal/users/users.go

```diff
@@ -161,6 +169,91 @@ func computeGroupOperation(newGroupsInfo []GroupInfo, currentLocalGroups []strin
 	return groupsToAdd, groupsToRemove
 }
 
+// CleanupSystemGroups removes all unexistent users from the system groups.
+//
+// In order to do that, it needs to query NSS for all existent users and then look at /etc/groups.
+func CleanupSystemGroups(_ context.Context, args ...Option) (err error) {
+	defer decorate.OnError(&err, "could not clean system groups completely")
+
+	defaultOptionsMu.RLock()
```

Comment:
I am unsure about the use of the mutex here, it doesn’t prevent data races when there are multiple tests in parallel unless you override all the default options.

Also, this was not needed for `UpdateLocalGroups` which has the exact same mechanism (and btw, you didn’t add the mutex there, while it’s modified with the others too).